ckan/lib/navl/dictization_functions.py

```diff
@@ -247,16 +247,14 @@ def convert(converter, key, converted_data, errors, context):
 
 def _remove_blank_keys(schema):
 
-    schema_copy = copy.copy(schema)
-
-    for key, value in schema.items():
+    for key, value in list(schema.items()):
         if isinstance(value[0], dict):
             for item in value:
                 _remove_blank_keys(item)
             if not any(value):
-                schema_copy.pop(key)
+                schema.pop(key)
```

Comment:
It's an error to modify a dict that you're iterating over

Reply:
"The reason that it works in python2.x is because kwargs.items() creates a list -- You can think of it as a snapshot of the dictionary's key-value pairs. Since it is a snapshot, you can change the dictionary without modifying the snapshot that you're iterating over and everything is OK.

In python3.x, kwargs.items() creates a view into the dictionary's key-value pairs. Since it is a view, you can no longer change the dictionary without also changing the view. This is why you get an error in python3.x

One resolution which will work on both python2.x and python3.x is to always create a snapshot using the list builtin"